CMakeLists.txt

```diff
@@ -58,11 +58,26 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/Modules/")
 
 
 # This is done here so that relative search paths are more reasonable
-find_package(Irrlicht)
-if(BUILD_CLIENT AND NOT IRRLICHT_FOUND)
-	message(FATAL_ERROR "IrrlichtMt is required to build the client, but it was not found.")
-elseif(NOT IRRLICHT_INCLUDE_DIR)
-	message(FATAL_ERROR "Irrlicht or IrrlichtMt headers are required to build the server, but none found.")
+if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/lib/irrlichtmt")
+	message(STATUS "Using user provided IrrlichtMt at subdirectory 'lib/irrlichtmt'")
+	# tell IrrlichtMt to create a static library
+	set(BUILD_SHARED_LIBS OFF CACHE BOOL "Build shared library" FORCE)
```

Comment:
this looks a big hacky but I guess there's no way around it
can you `unset()` the variable after including though?

Reply:
we aren't building any other libs, so this variable has no effect on anything else. But I'll try to reset it to its previous value

Reply:
Since we're not using (as you rightfully pointed out) just [unsetting](https://cmake.org/cmake/help/v3.20/command/unset.html) is fine.

Reply:
now the cache variable is unset after the `add_subdirectory()` call :heavy_check_mark: